omc/core/model/board.py

```diff
@@ -0,0 +1,545 @@
+from __future__ import annotations
+
+from typing import Any
+from typing import cast
+from typing import Collection
+from typing import MutableSet
+from typing import Optional
+
+import numpy as np
+from numpy import ndarray
+
+from omc.core.exception.exception import NegativePieceCoordinatesException
+from omc.core.exception.exception import NegativePiecePlayerNumException
+from omc.core.exception.exception import NonCurrentPiecePlayerNumException
+from omc.core.exception.exception import PieceAlreadyAtLocationException
+from omc.core.exception.exception import PiecePixelHexOutOfBoundsException
+from omc.core.exception.exception import PieceSubclassInvalidException
+
+
+class Board:
+    """
+    Object representing the state of the board, its present layout, and
+     dimensions.
+    """
+
+    def __init__(
+            self, dimensions: ndarray[int],
+            layout: ndarray[Player | None]
```

Comment:
Why is the layout an ndarray of Players, not Pieces?

Reply:
oops lol